### PR TITLE
Test pull request

### PR DESCRIPTION
**Pull request by akalinovski**

### CODE REVIEW

File: src/main/java/hello/HelloController.java [hunk 1]
```diff
@@ -36,4 +36,16 @@ public String index() {
     public String getUser() {
         return "John Doe";
     }
+
+    public Object clone() {
```
Comment (forsythe-cicd-pipeline) [inline src/main/java/hello/HelloController.java]: BLOCKER Remove this "clone" implementation; use a copy constructor or copy factory instead. [rule](https://sonarqube.api.cicd.siriuscloudservices.com/coding_rules#rule_key=squid%3AS2975)
MINOR Use super.clone() to create and seed the cloned instance to be returned. [rule](https://sonarqube.api.cicd.siriuscloudservices.com/coding_rules#rule_key=squid%3AS1182)

File: src/main/java/hello/HelloController.java [hunk 2]
```diff
@@ -36,4 +36,16 @@ public String index() {
...
+        return null;
+    }
+
+    public int compareTo(Object value) {
+        if (value == null) {
+            return Integer.MIN_VALUE;
```
Comment (forsythe-cicd-pipeline) [inline src/main/java/hello/HelloController.java]: MINOR Simply return -1 [rule](https://sonarqube.api.cicd.siriuscloudservices.com/coding_rules#rule_key=squid%3AS2167)

**Comment by forsythe-cicd-pipeline** [comment]
SonarQube analysis reported 3 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/hello/HelloController.java (hunk 1 above)
Comment (akrevenya) [reply src/main/java/hello/HelloController.java]: test comment from Alex

File: src/main/java/hello/HelloController.java [hunk 3]
```diff
@@ -1,16 +1,28 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import javax.servlet.http.HttpServletRequest;
 
 @RestController
-@Timed
-public class HelloController {
-    
+public class HelloController implements Comparable {
+    public static String FOO_VAR = "";
```
[the remark this answers is not in the document]
Comment (akrevenya) [reply src/main/java/hello/HelloController.java]: one more

**Comment by akalinovski** [comment]
Test

File: src/main/java/hello/HelloController.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (akrevenya) [reply src/main/java/hello/HelloController.java]: comment 2

**Comment by akalinovski** [comment]
Need more tests

File: src/main/java/hello/HelloController.java (hunk 1 above)
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: test comment

Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: test comment again

**Comment by akalinovski** [comment]
test comment

File: src/main/java/hello/HelloController.java (hunk 1 above)
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: New comment

File: src/main/java/hello/HelloController.java [hunk 4]
```diff
@@ -1,16 +1,28 @@
 package hello;
 
-import io.micrometer.core.annotation.Timed;
-import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import javax.servlet.http.HttpServletRequest;
```
Comment (akalinovski) [inline src/main/java/hello/HelloController.java]: Check import

Comment (akalinovski) [inline src/main/java/hello/HelloController.java]: Check again

Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Check again

File: src/main/java/hello/HelloController.java [hunk 5]
```diff
@@ -1,16 +1,28 @@
...
+    public static String FOO_VAR = "";
+
     @RequestMapping("/")
-    @Timed
-    public String index() {
+    public String index(HttpServletRequest request) {
         return "Greetings from Spring Boot!";
     }
+
+    public Object clone() {
```
Comment (akalinovski) [inline src/main/java/hello/HelloController.java]: check code

File: src/main/java/hello/HelloController.java (hunk 1 above)
Comment (akalinovski) [inline src/main/java/hello/HelloController.java]: Remove clone method

Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Remove again

File: src/main/java/hello/HelloController.java [hunk 6]
```diff
@@ -36,4 +36,16 @@ public String index() {
     public String getUser() {
         return "John Doe";
     }
+
+    public Object clone() {
+        return null;
+    }
+
+    public int compareTo(Object value) {
```
Comment (akalinovski) [inline src/main/java/hello/HelloController.java]: Check it again

File: src/main/java/hello/HelloController.java (hunk 1 above)
Comment (akalinovski) [inline src/main/java/hello/HelloController.java]: test comment again

File: src/main/java/hello/HelloController.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Wrong method

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Test

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Test

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Check

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Test

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Test

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Test

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Check

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: test

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Test

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Check

[the remark this answers is not in the document]
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Test

File: src/main/java/hello/HelloController.java (hunk 1 above)
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Check

Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: test

File: src/main/java/hello/HelloController.java (hunk 2 above)
Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: test

Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: test

Comment (akalinovski) [reply src/main/java/hello/HelloController.java]: Test

**Pull request closed without merging**